### PR TITLE
Now non-locatable data sources can create funcotations again.

### LINKED ISSUE #5773: Funcotator is only annotating with locatable funcotation factories
Currently Funcotator has an issue where it will only annotate with Locatable Funcotation Factories.

This is because of some logic in `DataSourceFuncotationFactory` around whether the `featureList` is compatible.

### PR DESCRIPTION
**Pull request by jonn-smith**
Primarily the large integration tests check for the non-locatable
funcotation factories producing data.

Regenerated the expected output from large tests.

Fixes #5773

### CODE REVIEW

File: scripts/funcotator/testing/testFuncotator.sh [hunk 1]
```diff
@@ -292,7 +292,7 @@ if [[ $r -eq 0 ]] && ${doRunLargeTests} ; then
     #INPUT=/Users/jonn/Development/gatk/src/test/resources/large/funcotator/regressionTestVariantSet2.vcf
     #INPUT=/Users/jonn/Development/gatk/src/test/resources/large/funcotator/regressionTestHg19Large.vcf
     #INPUT=/Users/jonn/Development/gatk/hg38_trio_liftoverb37.vcf
-    #INPUT=/Users/jonn/Development/gatk/tmp.vcf
+    INPUT=/Users/jonn/Development/gatk/tmp.vcf
```
Comment (LeeTL1220) [inline scripts/funcotator/testing/testFuncotator.sh]: Not sure why this is needed.  What is this stuff?

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/DataSourceFuncotationFactory.java [hunk 2]
```diff
@@ -135,6 +135,12 @@ public String getVersion() {
         return version;
     }
 
+    /**
+     * @return {@code True} if this {@link DataSourceFuncotationFactory} requires features to create {@link Funcotation}s.  {@code False} otherwise.
```
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/funcotator/DataSourceFuncotationFactory.java]: Any features?  Or specific types?  Please update docs.

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/DataSourceFuncotationFactory.java [hunk 3]
```diff
@@ -169,10 +175,20 @@ public String getVersion() {
...
+        final List<Funcotation> outputFuncotations;
+
         // Query this funcotation factory to get the list of overlapping features.
-        final List<Feature> featureList = queryFeaturesFromFeatureContext(featureContext);
+        // NOTE: This will only get features that are LOCATABLE!
+        //       This corresponds to requiresFeatures() returning `True`.
+        final List<Feature> featureList;
 
-        final List<Funcotation> outputFuncotations;
+        // Only get features if we need them:
```
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/funcotator/DataSourceFuncotationFactory.java]: Can you collapse to one line?  `blah? truestuff: falsestuff`

File: scripts/funcotator/testing/testFuncotator.sh (hunk 1 above)
Comment (jonn-smith) [reply scripts/funcotator/testing/testFuncotator.sh]: Left over from testing.  I'll remove.

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/DataSourceFuncotationFactory.java (hunk 2 above)
Comment (jonn-smith) [reply src/main/java/org/broadinstitute/hellbender/tools/funcotator/DataSourceFuncotationFactory.java]: Yes - any features.  If the DataSourceFuncotationFactory relies on feature matching that the GATK engine provides, then this should be true.

Things that don't require features (like gene name matching data sources) don't require any features to be present to create a funcotation.

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/DataSourceFuncotationFactory.java (hunk 3 above)
Comment (jonn-smith) [reply src/main/java/org/broadinstitute/hellbender/tools/funcotator/DataSourceFuncotationFactory.java]: Sure.

**Comment by jonn-smith** [comment]
Comments addressed.  I also added in another test to check for this specific case and guard against it in the future.

**Pull request merged**